File: archive/j/javascript/josephus-problem.js
const josephus = (n, k) => {
    if (n == 1) return 1;
    else return ((josephus(n - 1, k) + k - 1) % n) + 1;
};

const usage =
    "Usage: please input the total number of people and number of people to skip.";

const n = parseInt(process.argv[2]);
const k = parseInt(process.argv[3]);

if (!n || !k || typeof n !== "number" || typeof k !== "number") {
    return console.log(usage);
}

console.log(josephus(n, k));

File: archive/j/java/JosephusProblem.java
import java.util.*;

public class JosephusProblem {

    public static int josephus(int n, int k) {
        int res = 0;
        for (int i = 1; i <= n; i++) {
            res = (res + k) % i;
        }
        return res + 1;
    }

    public static void main(String[] args) {
        if (args.length != 2) {
            System.err.println("Usage: <n> <k>");
            System.exit(1);
        }

        try {
            int n = Integer.parseInt(args[0]);
            int k = Integer.parseInt(args[1]);
            if (n <= 0 || k <= 0) {
                System.err.println("Invalid input");
                System.exit(1);
            }
            System.out.println(josephus(n, k));
        } catch (NumberFormatException e) {
            System.err.println("Invalid input");
            System.exit(1);
        }
    }
}
